Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionTarget.kt

```diff
@@ -0,0 +1,146 @@
+
+    private fun onClickButtonVerifyHandler() {
+        val constraintLayout = binding.target.targetButtonVerify
+        val textView = binding.target.targetButtonVerify.target_verify
+        val initStateLayout: Array<ConstraintLayout> =
+            arrayOf(binding.target.targetButtonApply, binding.target.targetButtonReject)
+        val initStateTextView: Array<TextView> = arrayOf(
+            binding.target.targetButtonApply.target_apply,
+            binding.target.targetButtonReject.target_reject
+        )
+        val buttonNext = binding.submissionTargetButtonNext
+
+        changeState(constraintLayout, textView, initStateLayout, initStateTextView, buttonNext)
+    }
+
+    // onClickListener for targetButtonApply. Change style and enabled state for next Button
+
+    private fun onClickButtonApplyHandler() {
+        val constraintLayout = binding.target.targetButtonApply
+        val textView = binding.target.targetButtonApply.target_apply
+        val initStateLayout: Array<ConstraintLayout> =
+            arrayOf(binding.target.targetButtonReject, binding.target.targetButtonVerify)
+        val initStateTextView: Array<TextView> = arrayOf(
+            binding.target.targetButtonReject.target_reject,
+            binding.target.targetButtonVerify.target_verify
+        )
+        val buttonNext = binding.submissionTargetButtonNext
+
+        changeState(constraintLayout, textView, initStateLayout, initStateTextView, buttonNext)
+    }
+
+    // onClickListener for targetButtonReject. Change style and enabled state for next Button
+
+    private fun onClickButtonRejectHandler() {
+        val constraintLayout = binding.target.targetButtonReject
+        val textView = binding.target.targetButtonReject.target_reject
+        val initStateLayout: Array<ConstraintLayout> =
+            arrayOf(binding.target.targetButtonApply, binding.target.targetButtonVerify)
+        val initStateTextView: Array<TextView> = arrayOf(
+            binding.target.targetButtonApply.target_apply,
+            binding.target.targetButtonVerify.target_verify
+        )
+        val buttonNext = binding.submissionTargetButtonNext
+
+        changeState(constraintLayout, textView, initStateLayout, initStateTextView, buttonNext)
+    }
+
+    // change button's state - background and text color, when button is pressed or not, and enabled state for next button
+
+    fun changeState(
+        constraintLayout: ConstraintLayout,
+        textView: TextView,
+        initStateLayout: Array<ConstraintLayout>,
+        initStateText: Array<TextView>,
+        buttonNext: Button
+    ) {
+        val context = constraintLayout.context
+        var color = context.getColorStateList(R.color.colorGreenButtonNotPressed)
+        var textColor = context.getColor(R.color.colorTextGreenButtonNotPressed)
+        var enabledNext = false
+
+        if (constraintLayout.backgroundTintList === color) {
```

Comment:
Please don't check for a color, that doesn't make sense, we want to check for a state.